### DISCUSSION TITLE
Correct steps to update dependency, java-library org.apache.poi 4.1.1 to 5.2.3

### DISCUSSION THREAD

**Discussion by speckyspooky** [eclipse-birt/birt#1327] [Q&A] [answered]
Hello Together,

I want's to create a new branch to test the latest version of Apache POI which we use on BIRT for our excel (xlsx) implementation.
But I don't know why but maven and I'm not the best friends.

My question is I will write above my steps but I think I must have a brain-failure because the steps doesn't work.
I think I do something wrong and it would be nice if somebody can tell me the secret.

My steps to update the Apache POI 4.1.1 (current version) to 5.3.2:

1. checked the available library of org.apache.poi
=> result only: 4.1.1 is available like jar

2. checked the birt.target-file
=> 3 entries are given which must be updated:
- org.apache.poi
- org.apache.poi.ooxml
- org.apache.poi.ooxml.schemas
=> it could be that here later further updates are necessary depdendend on the used dependencies of 5.2.3

3. Open the Spudsoft-emitter bundle emitter & tests to change the MANIFEST.MF
=> I changed all 3 entries on "Require-Bundle" to:
- org.apache.poi;bundle-version="[5.2.3,6.0.0)",
- org.apache.poi.ooxml;bundle-version="[5.2.3,6.0.0)",
- org.apache.poi.ooxml.schemas;bundle-version="[5.2.3,6.0.0)",
=> Therefore the version 4.1.1 wasn't longer used

![2023-06-10 16_24_08-C__BIRT_dev_development_birt-master_git_master_build_org eclipse birt target_org](https://github.com/eclipse-birt/birt/assets/41593722/161272d9-e4e8-4313-a86e-08b7a294c8ce)

4. Maven Update
=> I started the Maven-Update-Project with forced-updates
=> The "updates" will be done also the project build but I didn't get the new version 5.3.2

![maven](https://github.com/eclipse-birt/birt/assets/41593722/75f85ff6-4a53-49ee-a05d-2d264c10d44d)

=> The MANIFEST.MF shows an error also after update (but the version number is fine, given on Maven repo)
![2023-06-10 16_36_46-ws - uk co spudsoft birt emitters excel_META-INF_MANIFEST MF - Eclipse IDE](https://github.com/eclipse-birt/birt/assets/41593722/f9e0a37e-0595-4aeb-be98-18232d40d9a1)

Where is my mistake on it?
Could somebody correct my steps or give me some hints for a correct update.

Many thanks in advance

**Comment by merks** [comment]
Saturday evening is not the best time for me to explain how things work...  

The target platform is **generated** like this:

![image](https://github.com/eclipse-birt/birt/assets/208716/4862331b-2c11-40b1-87f8-9dcf48b0371f)

If "we" want new versions of poi, "we're" going to have to look at how Orbit produces the wrappers for them.  Is there a compelling reason or need for this to happen sooner rather than later?    I.e., part of the correct steps are that it needs to be available as an OSGi bundle...

**Comment by merks** [comment, marked as answer]
FYI, this is where they come from currently and this is where they need to be updated to newer versions:

https://github.com/eclipse/orbit/tree/main/apache/org.apache.poi_4.1.1
https://github.com/eclipse/orbit/tree/main/apache/org.apache.poi.ooxml_4.1.1
https://github.com/eclipse/orbit/tree/main/apache/org.apache.poi.ooxml.schemas_4.1.1

Or they could be updated using some newer mechanism currently under discussion:

https://github.com/eclipse/orbit/issues/8

I would hope not to plan the release schedule around updates to poi.

**Reply by speckyspooky** [reply]
Ok, I thought there is a more independend way to refresh the dependency of Apache POI 5.2.3.
Therefore no, I doesn't plan to refresh the version for your current planned release 4.14 under those conditions.
Because it could be that there are some changes on the current excel-implementation are necessary - therefore I started the refresh try for my research.

But we should add this an update point for the next release because there are 2 major points to do it:
1. security findings - 6 noted CVEs (https://mvnrepository.com/artifact/org.apache.poi/poi/4.1.1)
2. major changes since publishing of 4.1.1 (Oct 2019)

If you have an idea how we can handle the orbit-change for the next release you are welocme!
(Because I doesn't know if we can change the orbit versions from our side or we need the orbit-project team.)

**Reply by merks** [reply]
I’m directly involved in helping to streamline Orbit so I will keep this in mind for poi.
